### PR TITLE
Add composition to Representable

### PR DESCRIPTION
**Pull request by TimWSpence**

### CODE REVIEW

**Comment by TimWSpence** [comment]
Realized I forgot to do `Nested` - will push that shortly

File: tests/src/test/scala/cats/tests/NestedSuite.scala [hunk 1]
```diff
@@ -264,4 +266,36 @@ class NestedSuite extends CatsSuite {
              SerializableTests.serializable(Distributive[Nested[Function1[Int, *], Function0, *]])
     )
   }
+
+  {
+    type Pair[A] = (A, A)
+
+    //Scala 2.12 implicit resolution absolutely loses its mind here
```
Comment (johnynek) [inline tests/src/test/scala/cats/tests/NestedSuite.scala:273]: do we need `implicit` if you are passing them explicitly below?

**Review by johnynek — COMMENTED** [review]
looks good to me other than a question about if we need some `implicit` modifiers.

File: tests/src/test/scala/cats/tests/NestedSuite.scala (hunk 1 above)
Comment (TimWSpence) [reply tests/src/test/scala/cats/tests/NestedSuite.scala:273]: Nope, good point. If we remove the modifier then we  could probably summon some of them via `implicitly` as well. Thanks, I'll fix this in both PRs 👍

**Review by johnynek — APPROVED** [review]

**Pull request merged**